app/models/user.rb

```diff
@@ -9,6 +9,8 @@ class User < ApplicationRecord
   has_many :projects, through: :project_users
   has_many :team_users
   has_many :teams, through: :team_users
-  has_many :comments
-  has_many :watchers
+  has_many :comments, dependent: :destroy
+  has_many :watchers, dependent: :destroy
+  has_many :created_issues, class_name: 'Issue', foreign_key: 'creator_id', inverse_of: 'creator', dependent: :destroy
+  has_many :assigned_issues, class_name: 'Issue', foreign_key: 'assignee_id', inverse_of: 'assignee', dependent: :nullify
```

Comment:
why nullify? discuss

Reply:
As the assignee for an issue can be null so if an assignee is deleted the assignee for issues for which he is the assignee should be set to null*.*